### PR TITLE
[EOC-71] add new list in cohort view

### PR DESCRIPTION
**Pull request by alemikolo**
cohort_view_with_add_list_button
cohort_view_with_creation_form
cohort_view_with_new_list

### CODE REVIEW

File: client/src/modules/cohort/index.jsx [hunk 1]
```diff
@@ -27,6 +38,41 @@ class Cohort extends PureComponent {
...
+    createList(name, description, userId, cohortId)
+      .then(this.hideListCreationForm)
+      .catch(noOp);
+  };
+
+  renderCreateListForm = () => {
+    const { listFormVisibility } = this.state;
+    return (
+      listFormVisibility && (
+        <CreationForm
```
Comment (marekrozmus) [inline client/src/modules/cohort/index.jsx]: create component for list form and put all functionality there - then use props

File: server/controllers/shoppingList.js [hunk 2]
```diff
@@ -2,22 +2,33 @@ const ShoppingList = require('../models/shoppingList.model');
...
-    err
-      ? resp.status(400).send({ message: err.message })
-      : resp
-          .location(`/shopping-list/${doc._id}`)
-          .status(201)
-          .send(doc);
+    if (err) {
+      return resp.status(404).send({ message: err.message });
+    }
+    if (!doc) {
```
Comment (marekrozmus) [inline server/controllers/shoppingList.js]: new lines between blocks

File: client/src/modules/dashboard/index.jsx [hunk 3]
```diff
@@ -77,40 +77,9 @@ class Dashboard extends Component {
...
-            onHide={this.hideForms}
-          />
-        </div>
-      )
-    );
-  };
-
   render() {
     const { shoppingLists, cohorts } = this.props;
+    const { cohortFormVisibility, shoppingFormVisibility } = this.state;
```
Comment (marekrozmus) [inline client/src/modules/dashboard/index.jsx]: change `shopping` to `list`

File: client/src/modules/cohort/index.jsx [hunk 4]
```diff
@@ -38,7 +71,21 @@ class Cohort extends PureComponent {
...
+            additionalIconSrc={PlusIcon}
+            mainIcon={<ListIcon />}
+            onClick={this.showListCreationForm}
+          >
+            <DropdownForm
+              isVisible={listFormVisibility}
+              label="Create new list"
+              onSubmit={this.createListSubmissionHandler}
+              type="menu"
+              onHide={this.hideListCreationForm}
```
Comment (adamklepacz) [inline client/src/modules/cohort/index.jsx]: sort alphabetically

File: client/src/modules/dashboard/index.jsx [hunk 5]
```diff
@@ -120,14 +89,26 @@ class Dashboard extends Component {
             mainIcon={<CohortIcon />}
             onClick={this.handleCohortFormVisibility}
           >
-            {this.renderCreateCohortForm()}
+            <DropdownForm
+              isVisible={cohortFormVisibility}
+              label="Create new cohort"
+              onSubmit={this.handleCohortSubmission}
+              type="menu"
+              onHide={this.hideForms}
```
Comment (adamklepacz) [inline client/src/modules/dashboard/index.jsx]: sort props

File: client/src/modules/dashboard/index.jsx [hunk 6]
```diff
@@ -120,14 +89,26 @@ class Dashboard extends Component {
...
-            onClick={this.handleShoppingListFormVisibility}
+            onClick={this.handleListFormVisibility}
           >
-            {this.renderCreateListForm()}
+            <DropdownForm
+              isVisible={listFormVisibility}
+              label="Create new list"
+              onSubmit={this.handleListSubmission}
+              type="menu"
+              onHide={this.hideForms}
```
Comment (adamklepacz) [inline client/src/modules/dashboard/index.jsx]: sort props

File: server/controllers/shoppingList.js [hunk 7]
```diff
@@ -2,22 +2,35 @@ const ShoppingList = require('../models/shoppingList.model');
...
-      ? resp.status(400).send({ message: err.message })
-      : resp
-          .location(`/shopping-list/${doc._id}`)
-          .status(201)
-          .send(doc);
+    if (err) {
+      return resp.status(404).send({ message: err.message });
+    }
+
+    if (!doc) {
```
Comment (marekrozmus) [inline server/controllers/shoppingList.js]: most probably not needed - if no error occured than we should have document

**Pull request merged**